### PR TITLE
Erlang syntax highlight

### PR DESCRIPTION
**Pull request by wingyplus**
This PR adds Erlang syntax highlight by using [`tree-sitter-erlang`](https://github.com/WhatsApp/tree-sitter-erlang) as a grammar. The highlights query write by hand instead of copying from the source because of I'm not sure it can copy due to LICENSE.

This PR didn't include indent, outline or LSP to scope down the PR size. But I can continue on later PR.  

Release Notes:

- Added Erlang syntax highlights ([#4939](https://github.com/zed-industries/zed/issues/4939)).

### CODE REVIEW

File: crates/zed/src/languages/erlang/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            "-" @keyword
+            name: (atom) @keyword))
+
+[
+    "("
+    ")"
+    "{"
+    "}"
+    "["
+    "]"
```
Comment (wingyplus) [inline crates/zed/src/languages/erlang/highlights.scm:96]: I'm not sure what indent size should use but it can be adjust.

**Comment by wingyplus** [comment]
@cla-bot check

**Comment by SomeoneToIgnore** [comment]
Hello, could you also add a docs entry please, into the https://github.com/zed-industries/zed/tree/main/docs/src/languages list.

File: assets/icons/file_icons/file_types.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
         "doc": "document",
         "docx": "document",
         "eex": "elixir",
+        "erl": "erlang",
```
Comment (wingyplus) [inline assets/icons/file_icons/file_types.json:25]: I found the issue with this file while editing with stable Zed is it trying to re-format the JSON file to 2 spaces indent. So I temporarily fix by editing outside the Zed to keep the indent.

**Comment by wingyplus** [comment]
> Hello, could you also add a docs entry please, into the https://github.com/zed-industries/zed/tree/main/docs/src/languages list.

Hi, the documentation has been added. Do you think adding only the tree-sitter is enough?

File: crates/zed/src/languages/erlang/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    "fun"
+    "if"
+    "maybe"
+    "of"
+    "receive"
+    "try"
+    "when"
+] @keyword
+
+; Term Comparisons
```
Comment (wingyplus) [inline crates/zed/src/languages/erlang/highlights.scm:22]: This should be removed.

**Comment by SomeoneToIgnore** [comment]
Sure, we have a few languages that do that, but what is missing here are outlines.scm that allow to open outlines menu (cmd-shift-o) and to enable breadcrumbs on top.

brackets.scm are used for bracket highlights and those seem to be missing, despite the language config.toml, I think it's not the way it should be.

**Comment by codeadict** [comment]
There is outline support in https://github.com/zed-industries/zed/pull/7093 if you wanna copy it @wingyplus

**Comment by SomeoneToIgnore** [comment]
Oh and https://github.com/zed-industries/zed/pull/7093 brings in the same grammar + LSP support, which seems more interesting: we think we should start closing just the syntax support PRs due to https://github.com/zed-industries/zed/pull/7105 and follow-ups is on track to provide tree-sitter highlights as plugins, generic for any language.

**Comment by wingyplus** [comment]
I think #7093 is more complete than my PR. Thanks to @codeadict. I think we can close this PR and continue the effort on #7093 instead. :)

**Comment by codeadict** [comment]
@SomeoneToIgnore Any plans to take a look at https://github.com/zed-industries/zed/pull/7093?

**Pull request closed without merging**